rqt_py_common/src/rqt_py_common/topic_helpers.py

```diff
@@ -28,161 +28,168 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
-import roslib
-import roslib.msgs
-import roslib.message
-from rostopic import get_topic_type
-from python_qt_binding.QtCore import qDebug
+# Author: Michael Lautman
 
+from ament_index_python import get_resource
 
-def get_type_class(type_name):
-    if roslib.msgs.is_valid_constant_type(type_name):
-        if type_name == 'string':
-            return str
-        elif type_name == 'bool':
-            return bool
-        else:
-            return type(roslib.msgs._convert_val(type_name, 0))
-    else:
-        return roslib.message.get_message_class(type_name)
+from rclpy import logging
 
 
-def get_field_type(topic_name):
+def get_topic_names_and_types(node=None):
```

Comment:
@dirk-thomas This feels like a bad way to do this, initialize rclpy and create a node just to get the topic names and types but I don't know of a better way... your thoughts?

Reply:
Here's an example from ros2cli DirectNode. Creates and destroys a temporary node. It includes _enter_ and _exit_ methods so you can use with syntax.

https://github.com/ros2/ros2cli/blob/master/ros2cli/ros2cli/node/direct.py

Reply:
very cool